### ISSUE TITLE
RFC Use Posix drand48 instead of rand (on non-/dev/random/Win32 platforms).

### ISSUE THREAD

**Issue by jaykrell** [mono/mono#9013]
Mono uses:
 1. Win32
 2. /dev/random
 3. ANSI C 89 rand

If /dev/random is not widespread, suggest use the Posix drand48 or such as a superior alternative to 3.

**Comment by NattyNarwhal** [comment]
On modern BSDs, there exists `arc4random` and I believe Linux has an equivalent syscall - the problem glibc might not expose it due to some plumbing crap.

https://man.openbsd.org/drand48

https://man.openbsd.org/arc4random

(FWIW maintaining the port of a Unix lacking most of this, `/dev/urandom` fallback is what we would have to use.)

**Comment by jaykrell** [comment]
Agreed arc4random. If it is truly a syscall on Linux, i.e. in the kernel, then glibc is not particularly relevant imho, can always call syscalls w/o glibc wrapper.

**Comment by NattyNarwhal** [comment]
Linux has [getrandom](http://man7.org/linux/man-pages/man2/getrandom.2.html) instead. There's no libc wrapper in glibc yet though.

**Comment by jaykrell** [comment]
arc4random_buf appears to be pretty universal and adequate/best -- Linux, ios, BSD.
getrandom/getentropy are just as good but more platform-specific.
CoreFX imports will drive us to arc4random_buf.
